ethers-etherscan/src/lib.rs

```diff
@@ -0,0 +1,454 @@
+            .form(form)
+            .send()
+            .await?
+            .json()
+            .await?)
+    }
+
+    /// Execute a api GET query
+    async fn get_json<T: DeserializeOwned, Q: Serialize>(
+        &self,
+        query: &Q,
+    ) -> anyhow::Result<Response<T>> {
+        Ok(self
+            .client
+            .get(self.etherscan_api_url.clone())
+            .header(header::ACCEPT, "application/json")
+            .query(query)
+            .send()
+            .await?
+            .json()
+            .await?)
+    }
+
+    fn create_query<T: Serialize>(
+        &self,
+        module: &'static str,
+        action: &'static str,
+        other: T,
+    ) -> Query<T> {
+        Query {
+            apikey: Cow::Borrowed(&self.api_key),
+            module: Cow::Borrowed(module),
+            action: Cow::Borrowed(action),
+            other,
+        }
+    }
+
+    /// Submit Source Code for Verification
+    pub async fn submit_contract_verification(
+        &self,
+        contract: &VerifyContract,
+    ) -> anyhow::Result<Response<String>> {
+        let body = self.create_query("contract", "verifysourcecode", contract);
+        Ok(self.post_form(&body).await?)
+    }
+
+    /// Check Source Code Verification Status with receipt received from
+    /// `[Self::submit_contract_verification]`
+    pub async fn check_verify_status(
+        &self,
+        guid: impl AsRef<str>,
+    ) -> anyhow::Result<Response<String>> {
+        let mut map = HashMap::new();
+        map.insert("guid", guid.as_ref());
+        let body = self.create_query("contract", "checkverifystatus", map);
+        Ok(self.post_form(&body).await?)
+    }
+
+    /// Returns the contract ABI of a verified contract
+    ///
+    /// ```no_run
+    /// # use ethers_etherscan::Client;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> anyhow::Result<()> {
+    ///     let client = Client::new("mainnet", "API_KEY").unwrap();
+    ///     let abi = client
+    ///         .contract_abi("0xBB9bc244D798123fDe783fCc1C72d3Bb8C189413".parse().unwrap())
+    ///         .await?;
+    ///
+    /// #     Ok(())
+    /// # }
+    /// ```
+    pub async fn contract_abi(&self, address: Address) -> anyhow::Result<Abi> {
+        let mut map = HashMap::new();
+        map.insert("address", address);
+        let query = self.create_query("contract", "getabi", map);
+        let resp: Response<String> = self.get_json(&query).await?;
+        Ok(serde_json::from_str(&resp.result)?)
+    }
+
+    /// Get Contract Source Code for Verified Contract Source Codes
+    /// ```no_run
+    /// # use ethers_etherscan::Client;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> anyhow::Result<()> {
+    ///     let client = Client::new("mainnet", "API_KEY").unwrap();
+    ///     let meta = client
+    ///         .contract_source_code("0xBB9bc244D798123fDe783fCc1C72d3Bb8C189413".parse().unwrap())
+    ///         .await?;
```

Comment:
Should we also compile this with Solc? Or provide some helper way to convert this to a `CompiledContract`

Reply:
the etherscan api doesn't seem to provide the bytecode directly, however it is present in the html reponse with `div[id=verifiedbytecode2]` -.-
Additional tooling to instantly compile this is probably useful, I'll add this to the tracking issue